clients/cobol-lsp-vscode-extension/src/services/copybook/downloader/ZoweExplorerDownloader.ts

```diff
@@ -86,16 +90,20 @@ export abstract class ZoweExplorerDownloader {
       this.storagePath,
     );
 
-    if (!(await DownloadUtil.checkPathExists(copybookPath))) {
-      try {
-        await this.downloadCopybookContent(dataset, member, profileName);
-        return true;
-      } catch (err: any) {
-        vscode.window.showErrorMessage(err.message);
-      }
+    const queueResponse = this.ZoweDownloadQueue.get(copybookPath);
+    if (queueResponse) {
+      return queueResponse;
     }
-
-    return false;
+    const response = this.downloadCopybookContent(
+      dataset,
+      member,
+      profileName,
+    ).catch((err) => {
```

Comment:
This will introduce caching, to preserve the original behavior use `finally`.
Also the path is no longer checked?

Reply:
updated